wp-includes/widgets/class-wp-widget-media.php

```diff
@@ -101,20 +102,25 @@ public function widget( $args, $instance ) {
 	 * @see WP_Widget::update()
 	 *
 	 * @param array $new_instance Values just sent to be saved.
-	 * @param array $old_instance Previously saved values from database.
+	 * @param array $instance     Previously saved values from database.
 	 * @return array Updated safe values to be saved.
 	 */
-	public function update( $new_instance, $old_instance ) {
-		$instance = $old_instance;
+	public function update( $new_instance, $instance ) {
 
 		// ID and title.
 		$instance['id']    = (int) $new_instance['id'];
 		$instance['title'] = sanitize_text_field( $new_instance['title'] );
 
-		// Everything else.
-		$instance['align'] = sanitize_text_field( $new_instance['align'] );
-		$instance['size']  = sanitize_text_field( $new_instance['size'] );
-		$instance['link']  = sanitize_text_field( $new_instance['link'] );
+		if ( in_array( $new_instance['align'], array( 'none', 'left', 'right', 'center' ) ) ) {
+			$instance['align'] = 'align' . $new_instance['align'];
```

Comment:
Right, but I was just trying to think of an example. Like, when we expose these widgets in the REST API, I'd surely hope we'd be exposing the data like:

```json
{"align": "left"}
```

Instead of:

```json
{"align": "alignleft"}
```

Also, more importantly, in the sanitization logic here:

```php
if ( in_array( $new_instance['align'], array( 'none', 'left', 'right', 'center' ) ) )
```

This means that a sanitized `align` instance value of `alignleft` would itself fail this sanitization check, as `alignleft` would not pass the `in_array` check, and it wouldn't be possible to take an existing instance and say copy it onto another widget.